### PR TITLE
Rust contract environment improvements

### PR DESCRIPTION
**Pull request by hatskier**

### CODE REVIEW

File: rust/run.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ async function main() {
 
 async function testInteraction(interaction) {
   console.log("\n\nExecuting interaction:", interaction);
-  const executionResult = await smartweaveContract.handle(interaction);
+  const executionResult = await smartweaveContract.handle_entrypoint(interaction);
```
Comment (ppedziwiatr) [inline rust/run.js:118]: this will require a change in SDK (note to myself ;-))

File: rust/src/contract_utils/entrypoint.rs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            None
+        } else {
+            Some(JsValue::from_serde(&result).unwrap())
+        };
+    } else {
+        Some(JsValue::from_serde(&result).unwrap())
+    }
+}
+
+fn log_tx() {
```
Comment (ppedziwiatr) [inline rust/src/contract_utils/entrypoint.rs:84]: I believe we can now remove this..

File: rust/src/contract_utils/entrypoint.rs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+    log(&format!("Transaction id {}", Transaction::id()));
+    log(&format!("Transaction owner {}", Transaction::owner()));
+    log(&format!("Transaction target {}", Transaction::target()));
+}
+
+
+#[wasm_bindgen(js_name = initState)]
+pub fn init_state(state: &JsValue) {
+    log(&format!("Calling init state 2"));
```
Comment (ppedziwiatr) [inline rust/src/contract_utils/entrypoint.rs:100]: and all the logs from these functions..

File: rust/src/contract_utils/entrypoint.rs [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    // not sure if that's deterministic - which is very important for the execution network.
+    // TODO: perf - according to docs:
+    // "This is unlikely to be super speedy so it's not recommended for large payload"
+    // - we should minimize calls to from_serde
+    let current_state = STATE.with(|service| service.borrow().clone());
+    JsValue::from_serde(&current_state).unwrap()
+}
+
+#[wasm_bindgen()]
+pub fn lang() -> String {
```
Comment (ppedziwiatr) [inline rust/src/contract_utils/entrypoint.rs:120]: this function and the next one (`contract_type`) needs to be adopted to the latest changes in the SDK.
https://github.com/redstone-finance/redstone-contracts-wasm/commit/7920c50da1d56a6c1e617a0fc5372b53dff58891#diff-f79aada6c062906719b8c4ea9c22fb916d82fc34d72fb0ccd80190bd80355ce2

File: rust/run.js (hunk 1 above)
Comment (hatskier) [reply rust/run.js:118]: No need anymore. I renamed hi function back to `handle`

File: rust/src/contract_utils/entrypoint.rs (hunk 2 above)
Comment (hatskier) [reply rust/src/contract_utils/entrypoint.rs]: Ok

File: rust/src/contract_utils/entrypoint.rs (hunk 3 above)
Comment (hatskier) [reply rust/src/contract_utils/entrypoint.rs]: Ok

**Pull request merged**